### DISCUSSION TITLE
Consider additional offline routers (brouter/routino, for different/better routing paths)

### DISCUSSION THREAD

**Discussion by gerion0** [organicmaps/organicmaps#500] [Ideas]
To plan bike routes, I normally use QMapShack, which utilitizes different offline routers for it's route planning, mainly [Routino](http://www.routino.org/) and [BRouter](http://brouter.de/). Especially, BRouter is made for bike routing (and supports different bike types).

Maybe you like the idea to support these routers in Organic Maps, too. Routino needs an extra database, what might be annoying (maybe you could provide one together with the maps), but in general this could improve routing a lot I guess.

**Comment by dexic79** [comment]
I support this idea - specifically the BRouter. It is a fast calculating tool and makes the usage of older hardware more pleasant. I am using it in OSMAnd on a S4 mini and it is much faster than the calculations in OSMAnd "original" and Organic Maps. 

Cheers!

**Comment by frasersab** [comment]
What is it that you prefer about BRouters routing? Do you have an example of where given the same points organicmaps routes differently to BRouter? Perhaps we could improve the OrganicMaps router instead. I think it is nice how to it's all built in and you don't have download another map for routing

**Comment by dexic79** [comment]
I prefer a couple of things in BRouter.

1. it calculates by a magnitude faster than Organic Maps or OSMAnd.
2. you have much more granular options to choose your preferences. Take a look at www.bikerouter.de for what BRouter offers. All of the calculation options are available on the phone, too.

My main point is (1). The route calculation in Organic Maps on my S4 mini is very slow. We are talking the difference between 3 seconds on OSMAnd with BRouter and 83 seconds with Organic Maps for a route of 41 km. I think that this is significant even on newer smartphones and tablets.

The preferences in BRouter are not such a big point, while I think that Organic Maps should stick more to official cycling routes to avoid motorized traffic. This is not an absolute rule but in my area, it seems to ignore them very often. Shorter ways are often preferred, while I would take a longer way around to be more safe. Still, it is useful to give a direction when you don't know where you are. 

My preferences are especially to stick to cycling routes and avoid unsafe traffic. Then, I can add my approximate level of fitness with how much power (W) I can apply and the weight of my bike and me, so the calculation is much more accurate. There are some levels in BRouter that are very useful for bike riding calculations.

**Comment by Misalf-git** [comment]
Also, with BRouter one can create different routing profiles tailored to specific needs. 

Like a simple hiking profile that prefers respective paths instead of getting there by foot using just any path or road.

But it's more versatile than just that.

I used it in the past and, if I remember correctly, the profile configuration options are near limitless so that a vehicle profile that excludes roads with **height**, **width** or **weight** restrictions could be created. The vehicle's **max speed** could be taken into account. Prefer either **short** or **fast** route and so on.

**Comment by Shohreh** [comment]
I agree a different router would be a good thing: The one currently used in OM can gladly send cyclists on dangerous roads.

Offering shortest/safest modes like BRouter should improve things.

<img width="800" height="1205" alt="image" src="https://github.com/user-attachments/assets/8bcbd35a-acb1-4ed0-892d-e88c56ee43da" />

<img width="521" height="614" alt="image" src="https://github.com/user-attachments/assets/d55eb41d-c6c3-414f-9373-bee58b0d258f" />
